ambry-cloud/src/main/java/com/github/ambry/cloud/azure/AzureCloudDestinationSync.java

```diff
@@ -454,7 +454,62 @@ public CompletableFuture<Short> undeleteBlobAsync(BlobId blobId, short lifeVersi
   @Override
   public short updateBlobExpiration(BlobId blobId, long expirationTime, CloudUpdateValidator cloudUpdateValidator)
       throws CloudStorageException {
-    return 0;
+    Timer.Context storageTimer = azureMetrics.blobUpdateTTLLatency.time();
+    AzureBlobLayoutStrategy.BlobLayout blobLayout = azureBlobLayoutStrategy.getDataBlobLayout(blobId);
+    String blobIdStr = blobLayout.blobFilePath;
+    BlobProperties blobProperties = getBlobProperties(blobLayout);
+    Map<String, String> cloudMetadata = blobProperties.getMetadata();
+
+    // Below is the correct behavior. For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
+    // We should never hit this case however because ReplicaThread::applyUpdatesToBlobInLocalStore does all checks.
+    if (cloudMetadata.containsKey(CloudBlobMetadata.FIELD_DELETION_TIME)
+        && Long.parseLong(cloudMetadata.get(CloudBlobMetadata.FIELD_DELETION_TIME)) != Utils.Infinite_Time) {
+      String error = String.format("Cannot update TTL of %s since it is already deleted in cloud.", blobIdStr);
+      throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.ID_Deleted));
+    }
+
+    /*
+    // Legacy cloudBlobStore does not expect an exception. However,
+    // below is the correct behavior. For ref, look at BlobStore::updateTTL and ReplicaThread::applyTtlUpdate.
+    // We should never hit this case however because ReplicaThread::applyUpdatesToBlobInLocalStore does all checks.
+    if (cloudMetadata.containsKey(CloudBlobMetadata.FIELD_EXPIRATION_TIME)
+        && Long.parseLong(cloudMetadata.get(CloudBlobMetadata.FIELD_EXPIRATION_TIME)) == Utils.Infinite_Time) {
+      String error = String.format("TTL of %s is already updated in cloud.", blobIdStr);
+      throw new CloudStorageException(error, new StoreException(error, StoreErrorCodes.Already_Updated));
+    }
+    */
+
+    Map<String, String> newMetadata = new HashMap<>();
+    newMetadata.put(CloudBlobMetadata.FIELD_EXPIRATION_TIME, String.valueOf(Utils.Infinite_Time));
+    cloudMetadata.putAll(newMetadata);
+
+    // lifeVersion must always be present
```

Comment:
If possible, would you be able to add a short description of reason on why it must always be present? Because it seems like we are only updating life version when Delete or Undelete is called. But I may be missing something.

Reply:
Check cloudBlobMetadatatoMap above. lifeVersion is added on PUT.

Reply:
done